Disable collector in phoebe.spec

shunghsiyu opened the pull request:
This is based on top of @asmorodskyi work in #45.

I removed Python-related dependencies and disabled the collector in `phoebe.spec`. Hopefully this is all that's left to do to get the CI happy.

asmorodskyi committed:
collect_stats: copy _phobe lib into common place

Before this execution of collect_stats.py was possible only with custom
PYTHONPATH. Now you can run collect_stats without any extra setup

shunghsiyu committed:
Disable collector in phoebe.spec

The collect_stats.py is for metric collection, with is no needed in production
environment where the main purpose is running inference, thus we disable the
_phoebe Python module needed by collect_stats.py.

dcermak commented on dist/rpm/phoebe.spec:
```diff
@@ -54,7 +51,7 @@ to offer eventually the best possible setup.
 # export build flags manually if %%set_build_flags is not defined
 %{?!set_build_flags:export CFLAGS="%{optflags}"; export LDFLAGS="${RPM_LD_FLAGS}"}
-%meson -Dwith_tests=false -Dsystemd_system_unit_dir=%{_unitdir}
+%meson -Dwith_tests=false -Dsystemd_system_unit_dir=%{_unitdir} -Dcollector=disabled
```
Adding this option makes no difference for the resulting rpm (it's just faster to build and has less build time dependencies), so I'm in favor of this.

shunghsiyu commented:
Hmm...I think [build_rpm failure on Leap 15.2](https://github.com/SUSE/phoebe/runs/2446288091?check_suite_focus=true) may be due to the use of [`disabler`](https://mesonbuild.com/Reference-manual.html#disabler)?

```
DEBUG: Program python3 found: NO (disabled by: collector )
DEBUG: scripts/meson.build:1:0: ERROR: Module returned a value of unknown type.
DEBUG: A full log can be found at /builddir/build/BUILD/phoebe-0.1.1/build/meson-logs/meson-log.txt
DEBUG: error: Bad exit status from /var/tmp/rpm-tmp.ywxUuq (%build)
DEBUG:     Bad exit status from /var/tmp/rpm-tmp.ywxUuq (%build)
DEBUG: RPM build errors:
DEBUG: Child return code was: 1
INFO: EXCEPTION: [Error()]
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/mockbuild/trace_decorator.py", line 93, in trace
    result = func(*args, **kw)
  File "/usr/lib/python3.9/site-packages/mockbuild/util.py", line 600, in do_with_status
    raise exception.Error("Command failed: \n # %s\n%s" % (command, output), child.returncode)
```

Let me look into that.

shunghsiyu committed:
Create disabler() object explicitly

While meson's documentation states that the 'disabler' parameter is supported
since 0.49.0[1], version 0.54.2 used in openSUSE Leap 15.2 fails with the
following error.

    Program python3 found: NO (disabled by: collector )

    scripts/meson.build:1:0: ERROR: Module returned a value of unknown type.

As a work around, we create the disabler() object explicitly in our
script/meson.build instead of relying on 'find_installation()'.

1: https://mesonbuild.com/Python-module.html#find_installation

mvarlese commented:
> Hmm...I think [build_rpm failure on Leap 15.2](https://github.com/SUSE/phoebe/runs/2446288091?check_suite_focus=true) may be due to the use of [`disabler`](https://mesonbuild.com/Reference-manual.html#disabler)?
> 
> ```
> DEBUG: Program python3 found: NO (disabled by: collector )
> DEBUG: scripts/meson.build:1:0: ERROR: Module returned a value of unknown type.
> DEBUG: A full log can be found at /builddir/build/BUILD/phoebe-0.1.1/build/meson-logs/meson-log.txt
> DEBUG: error: Bad exit status from /var/tmp/rpm-tmp.ywxUuq (%build)
> DEBUG:     Bad exit status from /var/tmp/rpm-tmp.ywxUuq (%build)
> DEBUG: RPM build errors:
> DEBUG: Child return code was: 1
> INFO: EXCEPTION: [Error()]
> Traceback (most recent call last):
>   File "/usr/lib/python3.9/site-packages/mockbuild/trace_decorator.py", line 93, in trace
>     result = func(*args, **kw)
>   File "/usr/lib/python3.9/site-packages/mockbuild/util.py", line 600, in do_with_status
>     raise exception.Error("Command failed: \n # %s\n%s" % (command, output), child.returncode)
> ```
> 
> Let me look into that.

@shunghsiyu do you want to first look at the leap15.2 failure or are you happy to have your patch merged nonetheless?

shunghsiyu commented:
With workaround in d52f48232ea81f2fbeb08dd0fc14b51140d20a36 the failure on Leap 15.2 does not show up anymore :smile: 

Let me go ahead and merge this PR
